api/fastpm/cosmology.h

```diff
@@ -3,22 +3,72 @@ FASTPM_BEGIN_DECLS
 extern double HubbleConstant;
 extern double HubbleDistance;
 
+typedef enum {
+    FASTPM_GROWTH_MODE_LCDM = 0,
+    FASTPM_GROWTH_MODE_ODE = 1,
+} FastPMGrowthMode;
+
 struct FastPMCosmology {
-    double OmegaM;
-    double OmegaLambda;
+    double h;
+    double Omega_m;
+    double Omega_cdm;
+    double Omega_Lambda;
+    double T_cmb;
+    //double T_nu;     // todays neutrino temperature HARD CODED FOR NOW
+    double N_eff;
+    int N_nu;          // total number of neutrino species (massive and massless)
+    double m_ncdm[3];  // masses of massive neutrinos (ncdm) for now assume max of 3 ncdms
+    int N_ncdm;
+    
+    FastPMGrowthMode growth_mode;
+    FastPMFDInterp * FDinterp;   // FIXME:? Will just be NULL in the case of no ncdm
```

Comment:
Can we use a FDinterp[1] here to avoid needing to decide when and who is responsible for freeing FDinterp?  (Then it is the cosmology object.

Reply:
So you're saying let's carry around the FDInterp structure in the cosmology object, instead of just the pointer? Could this slow things down at all?

Reply:
Note that FastPMFDInterp contains pointers to gsl_interp objects (see FDinterp.h). Is this also something to avoid?